src/sentry/static/sentry/app/views/organizationIncidents/details/index.jsx

```diff
@@ -63,6 +69,27 @@ class OrganizationIncidentDetails extends React.Component {
       });
   };
 
+  handleStatusChange = () => {
+    const {
+      api,
+      params: {orgId, incidentId},
+    } = this.props;
+
+    const status = isOpen(this.state.incident)
+      ? INCIDENT_STATUS.CLOSED
+      : INCIDENT_STATUS.CREATED;
+
+    updateStatus(api, orgId, incidentId, status)
+      .then(() => {
+        this.setState(state => ({
```

Comment:
Can we optimistically update?

Reply:
Yes - I also changed `updateSubscription` to do the same